src/ia-bookmarks-list.js

```diff
@@ -43,20 +49,84 @@ export class IABookmarksList extends LitElement {
     }));
   }
 
+  emitSaveBookmark(bookmark) {
+    this.dispatchEvent(new CustomEvent('saveBookmark', {
+      detail: {
+        bookmark,
+      },
+    }));
+  }
+
+  emitDeleteBookmark(id) {
+    this.dispatchEvent(new CustomEvent('deleteBookmark', {
+      detail: {
+        id,
+      },
+    }));
+  }
+
+  emitBookmarkColorChanged({ detail }) {
+    const { bookmarkId, colorId } = detail;
+    this.dispatchEvent(new CustomEvent('bookmarkColorChanged', {
+      detail: {
+        bookmarkId,
+        colorId,
+      },
+    }));
+  }
+
+  emitAddBookmark() {
+    this.dispatchEvent(new CustomEvent('addBookmark'));
+  }
+
+  editBookmark(e, bookmark) {
+    this.emitEditEvent(e, bookmark);
+    this.editedBookmark = this.editedBookmark === bookmark ? {} : bookmark;
+  }
+
+  saveBookmark({ detail }) {
+    const { bookmark } = detail;
+    this.editedBookmark = {};
+    this.emitSaveBookmark(bookmark);
+  }
+
+  deleteBookmark({ detail }) {
+    const { id } = detail;
+    this.editedBookmark = {};
+    this.emitDeleteBookmark(id);
+  }
+
   bookmarkItem(bookmark) {
+    const editMode = this.editedBookmark.id === bookmark.id;
     return html`
       <li
         @click=${() => this.emitSelectedEvent(bookmark)}
         class=${classMap({ active: bookmark.id === this.activeBookmarkID })}
+        tabindex="0"
       >
         <img src=${bookmark.thumbnail} />
         <h4>Page ${bookmark.page}</h4>
-        <button @click=${e => this.emitEditEvent(e, bookmark)} title="Edit this bookmark"><ia-icon-edit-pencil></ia-icon-edit-pencil></button>
-        ${bookmark.note ? html`<p>${bookmark.note}</p>` : nothing}
+        <button @click=${e => this.editBookmark(e, bookmark)} title="Edit this bookmark"><ia-icon-edit-pencil></ia-icon-edit-pencil></button>
+        ${!editMode && bookmark.note ? html`<p>${bookmark.note}</p>` : nothing}
```

Comment:
QUESTION: Any need to HTML-escape user-entered data here (e.g. preventing XSS attacks)? If so, could benefit from a test for this as well.

Reply:
LitElement does that for us, so the data can be saved exactly as the user input it and escaping occurs during render. 
![LitElement_escaping](https://user-images.githubusercontent.com/39553/96568920-7f981600-1296-11eb-9f03-b1c3bd52297b.gif)